core/src/main/java/org/apache/calcite/sql/dialect/PostgisSqlDialect.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.dialect;
+
+import org.apache.calcite.avatica.util.Casing;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlDialect;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.util.RelToSqlConverterUtil;
+
+/**
+ * A <code>SqlDialect</code> implementation for the PostgreSQL database.
+ */
+public class PostgisSqlDialect extends PostgresqlSqlDialect {
+
+  public static final SqlDialect.Context DEFAULT_CONTEXT = SqlDialect.EMPTY_CONTEXT
+      .withDatabaseProduct(DatabaseProduct.POSTGIS)
+      .withIdentifierQuoteString("\"")
+      .withUnquotedCasing(Casing.TO_LOWER)
+      .withDataTypeSystem(POSTGRESQL_TYPE_SYSTEM);
+
+  public static final SqlDialect DEFAULT = new PostgisSqlDialect(DEFAULT_CONTEXT);
+
+  /** Creates a PostgresqlSqlDialect. */
+  public PostgisSqlDialect(Context context) {
+    super(context);
+  }
+
+  @Override public void unparseCall(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
```

Comment:
Since PostGIS is an extension of PostgreSQL, Can these changes be processed directly in PostgresqlSqlDialect?

Reply:
Yes they could, I also had mixed feeling about extending the postgresql dialect. My current reasoning comes from the fact that postgis is an extension that must be installed on the server and then activated in the database with `CREATE EXTENSION postgis;`. Having a separate dialect makes things more explicit in calcite and isolates postgis specific customizations in a dedicated `unparseCall` method.

Reply:
It would be clearer to have an independent Dialect, but I am worried that it will introduce a maintenance burden. 
At present, PostGIS does not have many unique functions.

Reply:
ST_UNION is the main case I had in mind that required to introduce a new function name (ST_UNARYUNION) in calcite. However, some other functions, such as ST_DUMP, may require a similar treatment. I guess that the list will grow as the dialect gets more systematically tested against postgis.

Reply:
Simplest thing seems to be to make Postgis dialect a subclass of Postgres dialect. There are many similarities, but also many differences.